File: src/main/java/org/chousik/collection/validators/ValidatorLocationName.java
package org.chousik.collection.validators;

import org.chousik.exception.InvalidDataError;

/**
 * Валидатор названия локации
 */
public class ValidatorLocationName implements Validator<String> {
    /**
     * Проверяет строку на пустоту
     *
     * @param s строка
     * @throws InvalidDataError если строка пуста
     */
    @Override
    public void valide(String s) throws InvalidDataError {
        if (s == null) {
            return;
        }
        if (s.isBlank() | s.isEmpty()) {
            throw new InvalidDataError();
        }
    }
}

File: src/main/kotlin/collection/validators/ValidatorLocationName.kt
package org.chousik.collection.validators

import org.chousik.exception.InvalidDataError


class ValidatorLocationName : IValidator<String?> {

    @Throws(InvalidDataError::class)
    override fun valide(t: String?) {
        if (t == null) {
            return
        }
        if (t.isBlank() or t.isEmpty()) {
            throw InvalidDataError()
        }
    }
}